### DISCUSSION TITLE
ETA on Terragrunt change detection?

### DISCUSSION THREAD

**Discussion by gitbluf** [terramate-io/terramate#446] [Q&A] [answered]
Hello guys,

Great work on the tool! Really appreciated.

I was wondering is there maybe an ETA on this? As I understood from the docs it is currently only looking for .tf file changes?

**Comment by katcipis** [comment]
Hi @gitbluf,

We are currently working on a generalization that will allow you to provide a list of files to watch on change detection. If any files on that list changes your stack will be marked as changed. This should be landing this week :smile:. Hopefully that will cover Terragrunt and other integration scenarios. We may also support Terragrunt integration seamlessly in the future, but there is no ETA on that for now.

**Comment by mariux** [comment]
> As I understood from the docs it is currently only looking for .tf file changes?

Terramate detects changes in any files (any file type) within a stack and within all stacks sub directories. 

Terramate also scans `.tf` files for modules and detects changes if a local module is used. This is done recursively also for local modules referencing other local modules.

As tiago mentioned, we currently add a feature to explicitly watch other files too for changes.
This would allow you to manually create a list of files that terragrunt depends on or write a script to automatically create this list of files based on parsing terragrunt files like [transcend-io/terragrunt-atlantis-config](https://github.com/transcend-io/terragrunt-atlantis-config) is doing for Atlantis.

In the future we will add native support for context aware change detection.

Change detection only works within a git repository.

**Comment by katcipis** [comment, marked as answer]
@gitbluf this new release includes change detection watching any files configured in a stack: https://github.com/mineiros-io/terramate/releases/tag/v0.1.12

Some docs [here](https://github.com/mineiros-io/terramate/blob/v0.1.12/docs/change-detection.md#arbitrary-files-change-detection). Let us know if it helps :-)
